x-pack/plugins/elastic_assistant/server/lib/langchain/tools/alert_counts/get_alert_counts_tool.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { SearchResponse } from '@elastic/elasticsearch/lib/api/types';
+import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import { KibanaRequest } from '@kbn/core-http-server';
+import { DynamicTool, Tool } from 'langchain/tools';
+
+import { getAlertsCountQuery } from './get_alert_counts_query';
+import { requestHasRequiredAnonymizationParams } from '../../helpers';
+import type { RequestBody } from '../../types';
+
+export const ALERT_COUNTS_TOOL_DESCRIPTION =
+  'Call this for the counts of last 24 hours of open alerts in the environment, grouped by their severity';
+
+export const getAlertCountsTool = ({
+  alertsIndexPattern,
+  esClient,
+  replacements,
+  request,
+}: {
+  alertsIndexPattern?: string;
+  esClient: ElasticsearchClient;
+  replacements?: Record<string, string>;
+  request: KibanaRequest<unknown, unknown, RequestBody>;
+}): Tool | null => {
+  if (!requestHasRequiredAnonymizationParams(request) || alertsIndexPattern == null) {
```

Comment:
WRT the discussion above about not passing around `request`, `requestHasRequiredAnonymizationParams` will need to be extracted and `request` isolated in support of arbitrary tool registration on plugin start (https://github.com/elastic/kibana/pull/172234).

Reply:
For ref, these `tools` will live in `Security Solution` once we support generic tool registration.